### PR TITLE
Add glob to create_ensembles and mention flox in doc.

### LINKED ISSUE #1080: run test against xarray main
### Generic Issue

Hello I just merged a major upgrade to xarray's groupby/resample. Can you run your tests to make sure everything passes please?

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

### PR DESCRIPTION
**Pull request by aulemahal**
### Pull Request Checklist:
- [x] This PR addresses an already opened issue (for bug fixes / features)
    - This PR fixes #1080
- [x] Tests for the changes have been added (for bug fixes / features)
  - [x] (If applicable) Documentation has been added / updated (for bug fixes / features)
- [x] HISTORY.rst has been updated (with summary of main changes)
  - [x] Link to issue (:issue:`number`) and pull request (:pull:`number`) has been added
- [x] `bumpversion patch` has been called on this branch
- [x] The relevant author information has been added to `.zenodo.json`

### What kind of change does this PR introduce?

* This adds the possibility to pass a glob string to `create_ensemble` instead of a list of path.
* Added `flox` to the upstream test suite.
* Added mentions of `flox` in the installation instructions.
* Removed deprecated extra deps example in installation instructions and move `clisops` to the extra section.

### Does this PR introduce a breaking change?
No.

### Other information:
This is a very minor change, more like a reason to trigger the github actions and run all tests, including the one against xarray's main branch.

### CODE REVIEW

**Comment by aulemahal** [comment]
Oh! Well @dcherian, it looks like `flox` and the main of xarray breaks xclim! It seems the error are related to a different treatment of nans and errors in the results precision. Could the new algorithm modify the results slightly?

**Comment by dcherian** [comment]
Yeah I had to make some tests a little looser w.r.t precision.

The other failures look real :( NaNs in the wrong place; grouping by cftime;

**Comment by dcherian** [comment]
I noticed in some places you're using `np.testing` to compare xarray objects. Should be nicer to use `xarray.testing` instead (it will compare coordinate locations too)

```
    def test_season(self, tasmin_series, calendar):
        ts = tasmin_series(np.zeros(360))
        ts = convert_calendar(ts, calendar, missing=0, align_on="date")
    
        miss = missing.missing_any(ts, freq="YS", season="MAM")
>       np.testing.assert_equal(miss, [False])
E       AssertionError: 
E       Items are not equal:
E        ACTUAL: <xarray.DataArray (time: 1)>
E       array([ True])
E       Coordinates:
E         * time     (time) datetime64[ns] 2001-01-01
E        DESIRED: [False]
```

File: docs/installation.rst [hunk 1]
```diff
@@ -12,20 +12,11 @@ To install xclim via pip, run this command in your terminal:
 
     $ pip install xclim
 
-To install xclim with spatial subsetting tools (`clisops`_), ensure you have the necessary system dependencies installed and run:
```
Comment (Zeitsperre) [inline docs/installation.rst:15]: If we remove this section, we should probably also modify the `gis` recipe. Given that our `upstream` tox recipe is starting to get really complicated, maybe we can extract that from tox and offer a new recipe (that includes `flox` and `clisops`)?

File: docs/installation.rst [hunk 2]
```diff
@@ -41,7 +32,18 @@ To install the xclim Anaconda binary, run this command in your terminal:
 .. code-block:: console
 
     $ conda install -c conda-forge xclim
-    $ conda install -c conda-forge clisops  # for subsetting and bias correction functions
+
+Extra dependencies
+------------------
+To improve performance of xclim, we highly recommend you also install `flox`. This package
+integrates into xarray and improves the performance of the grouping and resampling
+algorithms, especially when using `dask` on large datasets.
```
Comment (Zeitsperre) [inline docs/installation.rst:40]:
```suggestion
To improve performance of xclim, we highly recommend you also install `flox`. This package integrates into xarray and significantly improves the performance of the grouping and resampling algorithms, especially when using `dask` on large datasets.
```

File: docs/installation.rst [hunk 3]
```diff
@@ -41,7 +32,18 @@ To install the xclim Anaconda binary, run this command in your terminal:
...
+
+We also recommend using the tools in `clisops.core.subset` for spatial manipulation of
+geospatial data.
```
Comment (Zeitsperre) [inline docs/installation.rst:43]:
```suggestion
We also recommend using the tools in `clisops.core.subset` for spatial manipulation of geospatial data.
```

File: HISTORY.rst [hunk 4]
```diff
@@ -9,6 +9,7 @@ Contributors to this version:  Abel Aoun (:user:`bzah`), Pascal Bourgault (:user
 Announcements
 ^^^^^^^^^^^^^
 * `xclim` is now compliant with `PEP 563 <https://peps.python.org/pep-0563>`_. Python3.10-style annotations are now permitted. (:issue:`1065`, :pull:`1071`).
+* Works with xarray's flox-enabled `GroupBy` and `resample`.
```
Comment (Zeitsperre) [inline HISTORY.rst:12]:
```suggestion
* `xclim` is now fully compatible with xarray's `flox`-enabled ``GroupBy`` and ``resample`` operations. (:pull:`1081`).
```

File: docs/installation.rst (hunk 1 above)
Comment (aulemahal) [reply docs/installation.rst:15]: But the `[gis]` recipe doesn't exist.  haha, I forgot to mention it above, but it seems it was deprecated without updating this file!

File: xclim/ensembles/_base.py [hunk 5]
```diff
@@ -1,8 +1,9 @@
 """Ensembles creation and statistics."""
 from __future__ import annotations
 
+from glob import glob
```
[the remark this answers is not in the document]
Comment (aulemahal) [reply xclim/ensembles/_base.py:4]: Oh! I didn't know there was a difference. I even forgot the existence of `pathlib.glob`. Haha cool that I did the correct choice unknowingly.

File: docs/installation.rst (hunk 1 above)
Comment (Zeitsperre) [reply docs/installation.rst:15]: Haha, in that case, good call! I don't even recall when that happened.

**Comment by aulemahal** [comment]
Some errors are caused by the resampling of `cftime` arrays, I opened an issue : pydata/xarray#6613, But for now, we can simply call those with a `xr.set_options(use_flox=False)` context.

**Comment by aulemahal** [comment]
Most other errors seems to be because of pydata/xarray#6615:
Flox uses `np.add` for the `sum`operation. But `np.add` acts as a OR on boolean values, while `np.sum` implicitly converts booleans to integers.

Xclim extensively resamples and sums over boolean values. The older xclim code usually explicitly casted those boolean arrays to int by multiplying by 1. But the newer code assumes the implicit behaviour from numpy/xarray and these parts are now broken.

File: xclim/core/missing.py [hunk 6]
```diff
@@ -91,7 +91,7 @@ def is_null(da, freq, **indexer):
         if selected.time.size == 0:
             raise ValueError("No data for selected period.")
 
-        null = selected.isnull()
+        null = selected.isnull() * 1
```
Comment (dcherian) [inline xclim/core/missing.py:94]: Should be able to remove these with the latest version of `flox`

File: xclim/testing/tests/test_sdba/test_base.py [hunk 7]
```diff
@@ -109,7 +109,10 @@ def test_grouper_apply(tas_series, use_dask, group, n):
         window_dim="window"
     )
     if grouper.prop != "group":
-        exp = rolld.groupby(group).mean(dim=[win_grouper.dim, "window"])
+        # todo: Remove method keyword when issue flox#102 is solved.
```
Comment (dcherian) [inline xclim/testing/tests/test_sdba/test_base.py:112]: Can also undo this with latest `flox`

File: xclim/ensembles/_base.py [hunk 8]
```diff
@@ -318,7 +324,10 @@ def _ens_align_datasets(
             time = xr.decode_cf(ds).time
 
             if resample_freq is not None:
-                counts = time.resample(time=resample_freq).count()
+                opts = {"use_flox": False} if "use_flox" in xr.get_options() else {}
```
Comment (dcherian) [inline xclim/ensembles/_base.py:327]: `count` will work finr now. Mean won't/

**Comment by aulemahal** [comment]
@tlogan2000  This is ready for review.

**Review by tlogan2000 — APPROVED** [review]
This looks good to me!  @dcherian thank-you for this addition to xarray I think xclim peformance will benefit a lot from this change.

@aulemahal @dcherian just so it's clear in my mind : what is expected behavior going forward on something like  `(xr.DataArray() > thresh).resample(time='YS').sum(dim='time')`  ?  Will the booleans be cast to ints in the end with `flox`?

**Comment by dcherian** [comment]
> Will the booleans be cast to ints in the end with flox?

Yes it will promote bool to int before aggregating. (https://github.com/dcherian/flox/commit/01b1afe1471e8a111df8994cdc7a83405e39b7b5). You could require `flox >=0.5.3` to be extra sure.

**Review by Zeitsperre — APPROVED** [review]

**Pull request merged**